i3lock.c

```diff
@@ -348,27 +353,29 @@ static void handle_key_press(xcb_key_press_event_t *event) {
         return;
     }
 
-    if ((input_position + 8) >= sizeof(password))
-        return;
+    if (!(capslock_indicator && ksym == XKB_KEY_Caps_Lock)) {
```

Comment:
Are you trying to ignore XCB_KEY_Caps_Lock? If so, add it to the switch/case statement above, that will make the diff much cleaner.